diff --git a/node_modules/@react-native/gradle-plugin/settings.gradle.kts b/node_modules/@react-native/gradle-plugin/settings.gradle.kts
index 69f1fe2..26db885 100644
--- a/node_modules/@react-native/gradle-plugin/settings.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/settings.gradle.kts
@@ -12,8 +12,6 @@ pluginManagement {
   }
 }
 
-plugins { id("org.gradle.toolchains.foojay-resolver-convention").version("0.5.0") }
-
include(
    ":react-native-gradle-plugin",
    ":settings-plugin",
diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/build.gradle.kts
@@ -60,7 +60,7 @@ dependencies {
 // error first. Instead we produce a Java 11-compatible Gradle Plugin, so that AGP can print their
 // nice message showing that JDK 11 (or 17) is required first
 java { targetCompatibility = JavaVersion.VERSION_11 }
 
-kotlin { jvmToolchain(17) }
+kotlin { jvmToolchain(JavaVersion.current().majorVersion.toInt()) }
 
 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
diff --git a/node_modules/@react-native/gradle-plugin/settings-plugin/build.gradle.kts b/node_modules/@react-native/gradle-plugin/settings-plugin/build.gradle.kts
--- a/node_modules/@react-native/gradle-plugin/settings-plugin/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/settings-plugin/build.gradle.kts
@@ -44,7 +44,7 @@ dependencies {
 // error first. Instead we produce a Java 11-compatible Gradle Plugin, so that AGP can print their
 // nice message showing that JDK 11 (or 17) is required first
 java { targetCompatibility = JavaVersion.VERSION_11 }
 
-kotlin { jvmToolchain(17) }
+kotlin { jvmToolchain(JavaVersion.current().majorVersion.toInt()) }
 
 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
diff --git a/node_modules/@react-native/gradle-plugin/shared/build.gradle.kts b/node_modules/@react-native/gradle-plugin/shared/build.gradle.kts
--- a/node_modules/@react-native/gradle-plugin/shared/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/shared/build.gradle.kts
@@ -25,7 +25,7 @@ dependencies {
 
 java { targetCompatibility = JavaVersion.VERSION_11 }
 
-kotlin { jvmToolchain(17) }
+kotlin { jvmToolchain(JavaVersion.current().majorVersion.toInt()) }
 
 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
diff --git a/node_modules/@react-native/gradle-plugin/shared-testutil/build.gradle.kts b/node_modules/@react-native/gradle-plugin/shared-testutil/build.gradle.kts
--- a/node_modules/@react-native/gradle-plugin/shared-testutil/build.gradle.kts
+++ b/node_modules/@react-native/gradle-plugin/shared-testutil/build.gradle.kts
@@ -19,7 +19,7 @@ group = "com.facebook.react"
 dependencies { implementation(libs.junit) }
 
 java { targetCompatibility = JavaVersion.VERSION_11 }
 
-kotlin { jvmToolchain(17) }
+kotlin { jvmToolchain(JavaVersion.current().majorVersion.toInt()) }
 
 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
