diff --git a/node_modules/react-native-purchases/android/build.gradle b/node_modules/react-native-purchases/android/build.gradle
index f1d7dc0..b3a1ecf 100644
--- a/node_modules/react-native-purchases/android/build.gradle
+++ b/node_modules/react-native-purchases/android/build.gradle
@@ -41,7 +41,7 @@ android {
         disable 'GradleCompatible'
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
     }
 }
